src/sagemaker/lambda_helper.py

```diff
@@ -166,7 +169,10 @@ def update(self):
                         Architectures=self.architectures,
                     )
                 else:
-                    bucket = self.s3_bucket or self.session.default_bucket()
+                    bucket, key_prefix = s3.calculate_bucket_and_prefix(
+                        self.s3_bucket, None, self.session
```

Comment:
nit: if possible lets use named args for debugging help.

Reply:
makes sense